pkg/sql/plan/base_binder.go

```diff
@@ -715,7 +722,62 @@ func (b *baseBinder) bindFuncExprImplByAstExpr(name string, astArgs []tree.Expr,
 		args[idx] = expr
 	}
 
-	return bindFuncExprImplByPlanExpr(b.GetContext(), name, args)
+	// return bindFuncExprImplByPlanExpr(b.GetContext(), name, args)
+	// first look for builtin func
+	builtinExpr, err := bindFuncExprImplByPlanExpr(b.GetContext(), name, args)
+	if err == nil {
+		return builtinExpr, nil
+	} else if !strings.Contains(err.Error(), "not supported") {
+		return nil, err
+	}
+
+	// not a builtin func, look to resolve udf
+	cmpCtx := b.builder.compCtx
+	udfSql, err := cmpCtx.ResolveUdf(name, args)
+	if err != nil {
+		return nil, err
+	}
+
+	return bindFuncExprImplUdf(b, name, udfSql, astArgs, depth)
+}
+
+func bindFuncExprImplUdf(b *baseBinder, name string, sql string, args []tree.Expr, depth int32) (*plan.Expr, error) {
+	// replace sql with actual arg value
+	fmtctx := tree.NewFmtCtx(dialect.MYSQL, tree.WithQuoteString(true))
+	for i := 0; i < len(args); i++ {
+		args[i].Format(fmtctx)
+		sql = strings.Replace(sql, "$"+strconv.Itoa(i+1), fmtctx.String(), 1)
+		fmtctx.Reset()
+	}
+
+	// if does not contain SELECT, an expression. In order to pass the parser,
+	// add a temporary SELECT in front of it.
+
+	var expr *plan.Expr
+
+	if !strings.Contains(sql, "select") {
```

Comment:
i think you means : start with 'select'?

Reply:
:) Yes that's what I meant, I'll change it now.